core/src/main/java/org/infinispan/interceptors/distribution/BiasedScatteredDistributionInterceptor.java

```diff
@@ -38,18 +41,23 @@ public void inject(CommandAckCollector collector, BiasManager biasManager) {
    }
 
    @Override
-   protected CompletionStage<Map<Address, Response>> singleWriteOnRemotePrimary(Address target, DataWriteCommand command) {
-      Collector<Response> collector = commandAckCollector.createBiased(command.getCommandInvocationId().getId(),
-            target, command.getTopologyId());
-      rpcManager.sendTo(target, command, DeliverOrder.NONE);
-      return collector.getFuture().thenApply(value -> Collections.singletonMap(target, value));
+   protected CompletionStage<ValidResponse> singleWriteOnRemotePrimary(Address target, DataWriteCommand command, ResponseCollector<ValidResponse> unused) {
+      BiasedCollector collector = commandAckCollector.createBiased(command.getCommandInvocationId().getId(),
+            command.getTopologyId());
+      super.singleWriteOnRemotePrimary(target, command, collector);
+      return collector.getFuture();
    }
 
    @Override
-   protected CompletionStage<Map<Address, Response>> manyWriteOnRemotePrimary(Address target, WriteCommand command) {
-      Collector<Response> collector = commandAckCollector.createMultiTargetCollector(command.getCommandInvocationId().getId(), target, command.getTopologyId());
-      rpcManager.sendTo(target, command, DeliverOrder.NONE);
-      return collector.getFuture().thenApply(value -> Collections.singletonMap(target, value));
+   protected CompletionStage<ValidResponse> manyWriteOnRemotePrimary(Address target, WriteCommand command, ResponseCollector<ValidResponse> unused, MultiTargetCollector multiTargetCollector) {
```

Comment:
I think the unused parameters complicate things a bit, I'd rather inline the call to `super` so only the `multiTargetCollector` needs to be passed in. Same in `singleWriteOnRemotePrimary`. WDYT?